File: movies-api/src/main/java/com/ivanfranchin/moviesapi/userextra/UserExtraService.java
package com.ivanfranchin.moviesapi.userextra;

import com.ivanfranchin.moviesapi.userextra.exception.UserExtraNotFoundException;
import com.ivanfranchin.moviesapi.userextra.model.UserExtra;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

import java.util.Optional;

@RequiredArgsConstructor
@Service
public class UserExtraService {

    private final UserExtraRepository userExtraRepository;

    public UserExtra validateAndGetUserExtra(String username) {
        return getUserExtra(username).orElseThrow(() -> new UserExtraNotFoundException(username));
    }

    public Optional<UserExtra> getUserExtra(String username) {
        return userExtraRepository.findById(username);
    }

    public UserExtra saveUserExtra(UserExtra userExtra) {
        return userExtraRepository.save(userExtra);
    }
}

File: movies-api/src/main/kotlin/com/ivanfranchin/moviesapi/userextra/UserExtraService.kt
package com.ivanfranchin.moviesapi.userextra

import com.ivanfranchin.moviesapi.userextra.exception.UserExtraNotFoundException
import com.ivanfranchin.moviesapi.userextra.model.UserExtra
import org.springframework.stereotype.Service
import java.util.*

@Service
class UserExtraService(
    private val userExtraRepository: UserExtraRepository
) {

    fun validateAndGetUserExtra(username: String): UserExtra = getUserExtra(username).orElseThrow {
        UserExtraNotFoundException(
            username
        )
    }

    fun getUserExtra(username: String): Optional<UserExtra> = userExtraRepository.findById(username)

    fun saveUserExtra(userExtra: UserExtra): UserExtra = userExtraRepository.save(userExtra)

}
